crates/sui-json-rpc/src/coin_api.rs

```diff
@@ -48,22 +47,30 @@ impl CoinReadApi {
     ) -> anyhow::Result<CoinPage> {
         let limit = cap_page_limit(limit);
         self.metrics.get_coins_limit.report(limit as u64);
-        let coins = self
-            .state
-            .get_owned_coins_iterator_with_cursor(owner, cursor, limit + 1, one_coin_type_only)?
-            .map(|(coin_type, obj_id, coin)| (coin_type, obj_id, coin));
-
-        let mut data = coins
-            .map(|(coin_type, coin_object_id, coin)| SuiCoin {
-                coin_type,
-                coin_object_id,
-                version: coin.version,
-                digest: coin.digest,
-                balance: coin.balance,
-                locked_until_epoch: None,
-                previous_transaction: coin.previous_transaction,
-            })
-            .collect::<Vec<_>>();
+        let state = self.state.clone();
+
+        let mut data = spawn_monitored_task!(async move {
```

Comment:
What is the reason to do `spawn_monitored_task(...).await` here and in similar places? The only effect of this is that the read request will continue to execute if RPC is cancelled, but I can't think why would we want this?

Reply:
Without this, the rocksdb call will block the jsonrpsee handle_request, this causes the RPC server to not respond to any new request until it complete.